language/move-compiler/src/parser/mod.rs

```diff
@@ -63,6 +63,8 @@ pub(crate) fn parse_program(
 
     let targets = find_move_filenames_with_address_mapping(targets)?;
     let mut deps = find_move_filenames_with_address_mapping(deps)?;
+    // Remove bytecode files
+    deps.retain(|p| !p.path.as_str().ends_with(MOVE_COMPILED_EXTENSION));
```

Comment:
Could you explain the reasoning to remove bytecode files from deps? 

Targets means that the compiler is intended to produce bytecode modules for those sources files, which I'm not sure makes sense here

Reply:
At this point, deps may already contain paths to mv files that are used as external dependencies, which will cause panic later when calling the function parse_file.

Reply:
So the "interface" files have been generated at this point right? Would it make sense to just remove the `.mv` files from deps after generating the source files for them?

Reply:
Thanks for the advice Todd! I moved the code to [here]( https://github.com/move-language/move/blob/4d24d3df379cf655eb43ef7fccd7db3e71472860/language/move-compiler/src/command_line/compiler.rs#L611). I am wondering whether it makes more sense?